torch/nn/utils/_expanded_weights/conv_utils.py

```diff
@@ -0,0 +1,184 @@
+import torch
+import torch.nn.functional as F
+
+import numpy as np
+from typing import List, Optional
+
+from .expanded_weights_utils import \
+    set_grad_sample_if_exists, unpack_expanded_weight_or_tensor
+
+THRESHOLD = 32
+
+def conv_picker(func, conv1dOpt, conv2dOpt, conv3dOpt):
+    if func == F.conv1d:
+        return conv1dOpt
+    if func == F.conv2d:
+        return conv2dOpt
+    else:
+        assert func == F.conv3d
+        return conv3dOpt
+
+conv_kwarg_defaults = {'bias': None, 'stride': 1, 'padding': 0, 'dilation': 1, 'groups': 1}
+def conv_args_and_kwargs(kwarg_names, expanded_args_and_kwargs):
+    args = expanded_args_and_kwargs[:len(expanded_args_and_kwargs) - len(kwarg_names)]
+    named_kwargs = expanded_args_and_kwargs[len(expanded_args_and_kwargs) - len(kwarg_names):]
+    unnamed_kwargs = args[2:]  # input and weight are the only two that can't be kwargs
+
+    conv_kwargs = tuple(conv_kwarg_defaults.keys())
+    kwargs = {key: value for (key, value) in zip(conv_kwargs[:len(unnamed_kwargs)] + kwarg_names, unnamed_kwargs + named_kwargs)}
+    kwargs = {key: kwargs.get(key, conv_kwarg_defaults[key]) for key in conv_kwarg_defaults}
+    return args[:2], kwargs
```

Comment:
I'm a little confused about the packing scheme. The reason I'm confused is that we get here, we are assuming that input and weight aren't kwargs.

If the user does something like F.conv2d(input=input, weight=weight, ...), does the `__torch_function__` trigger with input and weight as kwargs, or does `__torch_function__` normalize everything so that input & weight are args and everything else is a kwarg?

That being said, if the only way we're saying that people can hit expanded weights is through the module API (e.g. nn.Conv1d) then this isn't really a problem

Reply:
Yeah...you're totally right that if conv2d is just directly passing the args + kwargs through like we think it is, this won't work as well. My plan/proposal for the future is to have __torch_function__ normalize the inputs because that's what the norm functions do but it's probably something that will need a little more discussion

Reply:
Okay, I see two potential paths forward here:

The first is to just leave a TODO about needing an argument parser for `__torch_function__` and filing an issue about it. I don't understand how other people use `__torch_function__` without an argument parser...

The second is try to work around the issue by writing our own argument normalizer. It's not too difficult to do this, the challenge (for the long-term) is keeping it in sync with the actual signature for conv2d. For example, if we had a `args, kwargs = normalize_conv_inputs(input, weight, bias, ...)` etc function that has the same exact signature (including default arguments!) as conv2d, then that could be a quick workaround

```
def normalize_conv_inputs(input, weight, bias, ...):
  args = (input, weight, bias)
  kwargs = ...
  return args, kwargs
```

Reply:
Great idea for the short term! Added the second idea

For how other people use it, I found some other old issues too that I can pull. My gut is that we should parse it so it's passed to `__torch_function__` in a standardized manner but there's some old issues (it seems like from before `__torch_dispatch__`) so I want to investigate those a little more before changing the codegen